### ISSUE TITLE
[Bug]: It is possible to receive payments twice with same LN invoice

### ISSUE THREAD

**Issue by limpbrains** [synonymdev/bitkit-react-native#1488]
### Describe the bug

Using bitkit or WOS we were able to send to the same LN invoice a few times.

### Reproduce

1. Create LN invoice
2. Payit using different wallets
3. You will receive funds a few times

### Screenshots / Recording

receiver history:
![image](https://github.com/synonymdev/bitkit/assets/155891/09e94fa5-8502-4b07-a417-c060ed3f2409)

receiver payment1:
![image](https://github.com/synonymdev/bitkit/assets/155891/9d6eff97-73cc-4df2-bcc6-6c0cd640f73c)

receiver payment2:
![image](https://github.com/synonymdev/bitkit/assets/155891/c835e371-65a1-45aa-beaa-3ae35f89302c)

sender1 payment details:
![image](https://github.com/synonymdev/bitkit/assets/155891/9ea423c1-751b-48f3-8c89-9b62315208af)

sender2 payment details:
![image](https://github.com/synonymdev/bitkit/assets/155891/02ee07b7-db34-48b7-ba90-c35704ae4b09)

### Operating system

iOS, Android

### Bitkit version

v94

### Log output

[LDK logs from the receiver](https://github.com/synonymdev/bitkit/files/13918055/Zzzz.zip)

**Comment by limpbrains** [comment]
After restarting the app I only see one payment.
But before I did so, I was able to send 10 sats to another wallet

**Comment by JeanlChristophe** [comment]
This issue it's still not fix

**Comment by Jasonvdb** [comment]
Fix is [here](https://github.com/synonymdev/react-native-ldk/pull/236). Will update when in Bitkit.

**Comment by catch-21** [comment]
Tested on acf22eef1c63cad1dff84bc3304173825d01f6c0

The good: I haven't been able to pay an invoice twice, I always see an error.
The bad: The error is displayed later in the flow than it could be. It'd be good to reject the action at the point of input/scan.
The ugly: Presentation of the error message.


https://github.com/synonymdev/bitkit/assets/74595920/dc33090f-05c1-4963-91c6-14ae1b746394

**Comment by Jasonvdb** [comment]
So I agree with the bad and ugly part but this is a separate issue can be made for this. The fix I did [here](https://github.com/synonymdev/react-native-ldk/pull/236) was to prevent duplicate payments from the receiver side. It's always been blocked on the sender side when trying to pay same invoice twice. 

What was happening was 2 different lightning wallets were able to pay the same invoice, in this case they both thought they were the first so there was no check. So now with the fix the receiver enforces this check and only accepts the 1st payment.

**Comment by catch-21** [comment]
Yup, agreed. Raised https://github.com/synonymdev/bitkit/issues/1911.

Retested this using two different wallets to pay the same invoice and was not able to, so I'm happy to close this one.
